sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -201,6 +214,15 @@ public WritableColumnVector reserveDictionaryIds(int capacity) {
    */
   public abstract void putBooleans(int rowId, int count, boolean value);
 
+  /**
+   * Sets bits from [src[srcIndex], src[srcIndex + count]) to [rowId, rowId + count)
+   * src must be positive and contain 8 bits of bitmask in the lowest byte.
+   */
+  public void putBooleans(int rowId, int count, int src, int srcIndex) {
```

Comment:
why this takes a `int` for `src`, can we use `byte`?

Reply:
`int src` is more convenient and causes less number of casting.
Especially `int src` can prevent from from forgetting `& 0xFF`. If `byte src` we have to do `(long)src & 0xFF` because `byte src = 0xFF` will be `(long)src == 0xFFFFFFFFFFFFFFFFL` that is not convenient for `|` operations.

Anyway `in.read()` returns `int`, it is more natural to take an `int`

Reply:
Updated to `byte`